### PR TITLE
Enable VPA for alertmanager

### LINKED ISSUE #1789: Set elasticsearch logging VPA to auto
**What happened**:
`elasticsearch-logging-0` entered a crashloop backoff due to the `metrics-exporter` sidecar crashing and getting OOM killed. The `metrics-exporter` sidecar has static resources and does not get scaled. We should switch VPA to `auto` for this component to prevent situations like this in the future.
/cc @AndreasBurger

### PR DESCRIPTION
**Pull request by rfranzke**
**What this PR does / why we need it**:
With this PR we enable the VPA for `alertmanager`.
/cc @ialidzhikov @vlvasilev @wyb1 

**Release note**:

```improvement operator
The `alertmanager` `StatefulSet` is now automatically scaled vertically by the VPA.
```

### CODE REVIEW

File: charts/seed-bootstrap/charts/elastic-kibana-curator/templates/elasticsearch/elasticsearch-vpa.yaml [hunk 1]
```diff
@@ -9,4 +9,4 @@ spec:
     kind: StatefulSet
     name: elasticsearch-logging
   updatePolicy:
-    updateMode: "Off"
+    updateMode: "Auto"
```
Comment (vpnachev) [inline charts/seed-bootstrap/charts/elastic-kibana-curator/templates/elasticsearch/elasticsearch-vpa.yaml]: To make VPA for elastic to work, these flags must be adapted, too. 

https://github.com/gardener/gardener/blob/f85b9461857c5e6652c2b3258d57cf5bbe1e4fd7/charts/seed-bootstrap/charts/elastic-kibana-curator/templates/elasticsearch/statefulset.yaml#L47

**Review by vpnachev — CHANGES_REQUESTED** [review]
This is not enough to enable VPA for the Elastic.

File: charts/seed-bootstrap/charts/elastic-kibana-curator/templates/elasticsearch/elasticsearch-vpa.yaml (hunk 1 above)
Comment (vpnachev) [reply charts/seed-bootstrap/charts/elastic-kibana-curator/templates/elasticsearch/elasticsearch-vpa.yaml]: I have spent some time on this.
The used elastic is running on java 12
```bash
$ /opt/jdk-12/bin/java  -version                                                                                     
openjdk version "12" 2019-03-19
OpenJDK Runtime Environment (build 12+33)
OpenJDK 64-Bit Server VM (build 12+33, mixed mode, sharing)
```

For this version `UseContainerSupport` is enabled by default, see https://www.eclipse.org/openj9/docs/xxusecontainersupport/

To achieve better performance, I think we have to replace the `-Xms` and `-Xmx` flags with the more container friendly `InitialRAMPercentage`, `MaxRAMPercentage` and `MinRAMPercentage`, ref https://www.eclipse.org/openj9/docs/xxinitialrampercentage/. Or let use the default ones calculated on the detected resources, but remove the `-Xms` and `-Xmx`, for sure.
```bash
$ /opt/jdk-12/bin/java  -XX:+PrintFlagsFinal -version | grep RAMPer                                                  
   double InitialRAMPercentage                     = 1.562500                                  {product} {default}                                    
   double MaxRAMPercentage                         = 25.000000                                 {product} {default}                                    
   double MinRAMPercentage                         = 50.000000                                 {product} {default}
```

Ref:
https://merikan.com/2019/04/jvm-in-a-container/#java-10

Comment (rfranzke) [reply charts/seed-bootstrap/charts/elastic-kibana-curator/templates/elasticsearch/elasticsearch-vpa.yaml]: @vpnachev thanks for digging into this. I've removed the flags. Can you check again, please?

Comment (vpnachev) [reply charts/seed-bootstrap/charts/elastic-kibana-curator/templates/elasticsearch/elasticsearch-vpa.yaml]: Unfortunately, this is not enough. 
You have to remove them from here, too. https://github.com/gardener/gardener/blob/cc4dd233de46aff4f69b3aedffcb91817e766e16/charts/seed-bootstrap/charts/elastic-kibana-curator/templates/elasticsearch/configmap.yaml#L129-L130

Comment (rfranzke) [reply charts/seed-bootstrap/charts/elastic-kibana-curator/templates/elasticsearch/elasticsearch-vpa.yaml]: Thanks, done.

File: charts/seed-bootstrap/charts/elastic-kibana-curator/templates/elasticsearch/configmap.yaml [hunk 2]
```diff
@@ -126,9 +126,6 @@ data:
     # Xms represents the initial size of total heap space
     # Xmx represents the maximum size of total heap space
```
Comment (swilen-iwanow) [inline charts/seed-bootstrap/charts/elastic-kibana-curator/templates/elasticsearch/configmap.yaml]:
```suggestion
    -XX:MaxRAMPercentage=50.0
    -XX:MinRAMPercentage=50.0
    -XX:InitialRAMPercentage=50.0
```

**Review by swilen-iwanow — CHANGES_REQUESTED** [review]
The Elastic requires min and max heap to be equal, thus we cannot use the defaults.

PS. We still have not seen the VPA autoscaling in action, will need more time.

**Comment by rfranzke** [comment]
I changed the PR for now to only activate VPA for the alertmanager. @swilen-iwanow thanks for the suggestions, I've assign you to #1789 as you are looking into it, right? If not please unassign yourself again. Thanks.

**Review by wyb1 — APPROVED** [review]

**Pull request merged**